sql/analyzer/fix_field_indexes.go

```diff
@@ -63,13 +68,18 @@ func FixFieldIndexes(scope *Scope, schema sql.Schema, exp sql.Expression) (sql.E
 				offset += len(schema)
 				for i, col := range schema {
 					if e.Name() == col.Name && e.Table() == col.Source {
-						return expression.NewGetFieldWithTable(
-							scopeLen-offset+i,
-							e.Type(),
-							e.Table(),
-							e.Name(),
-							e.IsNullable(),
-						), nil
+						newIndex := scopeLen - offset + i
+						if e.Index() != newIndex {
+							a.Log("Rewriting field %s.%s from index %d to %d", e.Table(), e.Name(), e.Index(), newIndex)
```

Comment:
seems like a lot of threading to get the log statement here

Reply:
Yeah, eventually we need to put the logger in the session or something